storage/storage_transport_test.go

```diff
@@ -0,0 +1,137 @@
+			require.True(t, ok, c.prefix)
+			assert.Equal(t, c.expectedDriver, storageRef.transport.store.GetGraphDriverName(), c.prefix)
+			assert.Equal(t, c.expectedRoot, storageRef.transport.store.GetGraphRoot(), c.prefix)
+		}
+	}
+}
+
+func TestTransportValidatePolicyConfigurationScope(t *testing.T) {
+	store := newStore(t)
+	driver := store.GetGraphDriverName()
+	root := store.GetGraphRoot()
+	storeSpec := fmt.Sprintf("[%s@%s]", driver, root) // As computed in PolicyConfigurationNamespaces
+
+	// Valid inputs
+	for _, scope := range []string{
+		"[" + root + "suffix1]",                                              // driverlessStoreSpec in PolicyConfigurationNamespaces
+		"[" + driver + "@" + root + "suffix3]",                               // storeSpec
+		storeSpec + "sha256:ab",                                              // Valid single-component name (ParseIDOrReference accepts digest prefixes as names!) (FIXME? is this desirable?)
+		storeSpec + "sha256:" + sha256digestHex,                              // Valid single-component ID with a tag
+		storeSpec + "busybox",                                                // Valid single-component name, implicit tag; NOTE that this non-canonical form would be interpreted as a scope for host busybox
+		storeSpec + "busybox:notlatest",                                      // Valid single-component name, explicit tag; NOTE that this non-canonical form would be interpreted as a scope for host busybox
+		storeSpec + "docker.io/library/busybox:notlatest",                    // Valid single-component name, everything explicit
+		storeSpec + "busybox@" + sha256digestHex,                             // Valid two-component name, implicit tag; NOTE that this non-canonical form would be interpreted as a scope for host busybox (and never match)
+		storeSpec + "busybox:notlatest@" + sha256digestHex,                   // Valid two-component name, explicit tag; NOTE that this non-canonical form would be interpreted as a scope for host busybox (and never match)
+		storeSpec + "docker.io/library/busybox:notlatest@" + sha256digestHex, // Valid two-component name, everything explicit
+	} {
+		err := Transport.ValidatePolicyConfigurationScope(scope)
+		assert.NoError(t, err, scope)
+	}
+
+	// Invalid inputs
+	for _, scope := range []string{
+		"busybox",                        // Unprefixed reference; FIXME: This can't actually be matched by a storageReference.PolicyConfiguration{Identity,Namespaces}, so it should be rejected
+		"[unterminated",                  // Unterminated store specifier
+		"[]",                             // Empty store specifier
+		"[relative/path]",                // Non-absolute graph root path
+		"[" + driver + "@relative/path]", // Non-absolute graph root path
+		// "[thisisunknown@" + root + "suffix2]", // Unknown graph driver FIXME? Should this be detected?
+		storeSpec + sha256digestHex,       // Valid single-component ID, implicit digest.Canonical, but ID-only
```

Comment:
This is not parsed as an ID any more; it is AFAICT rejected because repository names must not look like sha256 IDs (64 hex characters)